ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/storage/keyvalue/DatabaseMetadata.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.storage.keyvalue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class DatabaseMetadata {
+  private static final String METADATA_FILENAME = "DATABASE_METADATA.json";
```

Comment:
As far as I know our configuration files have all been `.toml` to date (auth, permissioning, CLI options).  How hard would it be to adapt this to a TOML format?

Reply:
Genesis files are JSON-based, and this is an internal metadata file rather than user-facing config so I don't think consistency is a big concern ...  Which is to say, I don't think its a big deal if we use JSON here.  I also think that `jackson` is a pretty good library to lean on and it doesn't support TOML.

Reply:
We use the Cava TOML parser in places like the whitelist persister - https://github.com/PegaSysEng/pantheon/blob/3fffd605dd065a2efbb8d793429713a3d9076d1d/ethereum/permissioning/src/main/java/tech/pegasys/pantheon/ethereum/permissioning/WhitelistPersistor.java#L89

The RocksDB config files in that directory, OPTIONS-#####, are also TOML based.  genesis.json seems to be the only outlier when it comes to files living on the file system.

Reply:
Unless I'm missing something, it looks like cava is archived? https://github.com/ConsenSys/cava

It's much defintely much cleaner in jackson than cava but I see your point about consistency :P

It seems our options are:
* bring in a dependency on a maintained toml library just for this config
  * then in another pr, migrate other areas over to that library, removing cava dependency
* use jackson

Reply:
Marking this resolved for now as cava doesn't have utilities to write toml. We would have to pull in another toml dependency that has writing or roll our own.

With just the version flag as of now, it would be relatively simple to write the string on our own but in case we end up using more of the toml grammar in the future, it could be unwieldy to maintain.